jinahub/encoder/image/clip.py

```diff
@@ -0,0 +1,80 @@
+from itertools import islice
+from typing import Optional, Iterable, Generator, Any
+
+import numpy as np
+import torch
+from jina import Executor, DocumentArray, requests
+
+import clip
+
+
+def _batch_generator(generator: Generator[int, Any, None], size: int):
```

Comment:
I think the `problem` here is that it does notr make sense `to gnerate` a `batch` from a `generator`. THe input should be a `list`. It does not make sense to provide `generator` as input because it is already in memory so no gain here.

It is the output that should be a `generator` and not the `input`

Reply:
is fine to me as well. I changed it to accept a list as input